docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,309 @@
+[[migrate_to_java_time]]
+=== Java Time Migration Guide
+
+Since version 7.0 Elasticsearch is using java.time as an underlying implementation of dates parsing, formatting and calculations.
+java.time is in many aspects very similar to joda-time which we previously used. However there are some differences and this guide is meant to
+help you to become compatible.
+
+==== Identifying if you are affected
+You are only affected if you have used any custom date formats in your mappings, pipelines or templates.
+Most of the custom date formats are compatible, but with some there is a change of meaning or missing. More on that below.
+A custom format is specified on a date field like this:
+[source,console]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "date": {
+        "type":   "date",
+        "format": "YYYY/MM/dd HH:mm:ss||YYYY/MM/dd||epoch_millis"
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+To make sure if any of the date formats is deprecated run deprecation API or kibana upgrade assistant.
+//todo fix this
+xx migration-api-deprecation xx
+xxinclude ::../apis/deprecation.asciidoc
+
+Sample output of deprecation api will be as follows:
+[source,text]
+--------------------------------------------------
+This index has date fields with deprecated formats: [[type: doc, field: param2, format: yyyy/MM/dd HH:mm:ss||yyyy/MM/dd||epoch_millis, suggestion: 'y' year should be replaced with 'u'. Use 'y' for year-of-era.]]. Prefix your date format with '8' to use the new specifier.
+--------------------------------------------------
+
+
+==== Incompatible formats
+If a pattern contains literals below, that means the pattern is incompatible and needs to be changed.
+
+===== Y - in 6.8 meaning Year of era. It should be replaced to y in 7.0 This is a big change because in 7.0 'Y' means week-based-year.
```

Comment:
Rather than using headings `=====`, I'd create a definition list with each literal:

`Y` (Year of era)::
Replace with `y`. In java time, `Y` is used for week-based year (e.g. `2020W52`).

+
Example:
`YYYY-MM-dd` should become `yyyy-MM-dd`

Reply:
good idea - fixed